portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,89 +20,116 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 import com.google.gson.Gson;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime; 
+import java.lang.Math;
 
-/** Servlet that handles comment data. */
+/** 
+  * Servlet that uploads and retrieves persistent comment data using datastore.
+  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Represents a single comment */
+  private static class Comment {
+      String text;
+      String name;
+      String time;
+
+      Comment(String text, String name, String time) {
+        this.text = text;
+        this.name = name;
+        this.time = time;
+      }
+  }
+
+  /** Extracts user comment from form and stores it via datastore. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String userComment = request.getParameter("text-input");
-    if (userComment != null) {
-      storeComment(userComment);
+    String userName = request.getParameter("author");
+    String timestamp = Long.toString(System.currentTimeMillis());
+
+    if (userComment != null || userComment.trim() == "") {
```

Comment:
Guava's Strings library is an incredibly useful tool that I found out about too late (in my opinion) in my life. Now, I use it a ton whenever I'm handling strings. 

Give it a look?

https://guava.dev/releases/19.0/api/docs/com/google/common/base/Strings.html

Reply:
Good idea! I'm now using Strings.isNullOrEmpty() for this